File: solution/1371. Find the Longest Substring Containing Vowels in Even Counts/1371_02.cpp
class Solution {
public:
    int findTheLongestSubstring(string s) {
        unordered_map<int, int> m{{0, -1}};
        
        int res = 0;
        for (int i = 0, state = 0; i < s.size(); ++i) {
            
            int idx = string("aeiou").find(s[i]);
            if (idx != -1)
                state ^= (1 << idx);
            
            if (m.count(state))
                res = max(res, i - m[state]);
            else
                m[state] = i;
        }
        return res;
    }
};

File: solution/1371. Find the Longest Substring Containing Vowels in Even Counts/1371_02.py
class Solution:
    def findTheLongestSubstring(self, s: str) -> int:
        seen = {0: -1}
        res = state = 0
        for i, c in enumerate(s):
            j = 'aeiou'.find(c)
            if j != -1:
                state ^= 1 << j
            seen.setdefault(state, i)
            res = max(res, i - seen[state])
        return res
    
